tuf/client_rework/updater_rework.py

```diff
@@ -0,0 +1,786 @@
+# Copyright 2020, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""TUF client 1.0.0 draft
```

Comment:
> Cool stuff, @sechkova! Is this as feature-complete as the existing `updater.py`? If so, nice job condensing it to 1/3 of LOC! :)

I believe it implements the same features as the existing updater but most likely it misses to address corner cases. Anyway, in terms of volume I think this will be approximately the end result :)

> 
> Some high-level comments:
> 
> * `_mirror_meta_download` and `_mirror_target_download` look a lot a like. Do we need both?

I consider them part of the "totally rework download and mirrors" plan and I hope to see them gone!

> * Same goes for `_get_full_meta_name` and `_get_relative_meta_name.` Maybe the former could be a wrapper around the latter?
> * The 5 top-level functions at the bottom of the module feel a bit lost. Should we add a client.util module? Or make them part of the `Updater` class, e.g. as `staticmethod`s?

Yes, you are right, `staticmethod` would have helped describe them better. They are actually copy-paste transfer of the old code which I needed to make a full update cycle work. They need a better place (same as `_get_*_name` methods)